scripts/tunnel-creation.mjs

```diff
@@ -0,0 +1,348 @@
+
+      log.info('Creating lockdown service...');
+      const {lockdownService, device: lockdownDevice} = await createLockdownServiceByUDID(udid);
+      log.info(`Lockdown service created for device: ${lockdownDevice.Properties.SerialNumber}`);
+
+      log.info('Starting CoreDeviceProxy...');
+      const {socket} = await startCoreDeviceProxy(
+        lockdownService,
+        lockdownDevice.DeviceID,
+        lockdownDevice.Properties.SerialNumber,
+        tlsOptions,
+      );
+      log.info('CoreDeviceProxy started successfully');
+
+      log.info('Creating tunnel...');
+      const tunnel = await TunnelManager.getTunnel(socket);
+      log.info(`Tunnel created for address: ${tunnel.Address} with RsdPort: ${tunnel.RsdPort}`);
+
+      let packetStreamPort;
+      packetStreamPort = this._packetStreamBasePort++;
+      const packetStreamServer = new PacketStreamServer(packetStreamPort);
+      await packetStreamServer.start();
+
+      const consumer = packetStreamServer.getPacketConsumer();
+      if (consumer) {
+        tunnel.addPacketConsumer(consumer);
+      }
+
+      this._packetStreamServers.set(udid, packetStreamServer);
+
+      log.info(`Packet stream server started on port ${packetStreamPort}`);
+
+      log.info(`✅ Tunnel creation completed successfully for device: ${udid}`);
+      log.info(`   Tunnel Address: ${tunnel.Address}`);
+      log.info(`   Tunnel RsdPort: ${tunnel.RsdPort}`);
+      if (packetStreamPort) {
+        log.info(`   Packet Stream Port: ${packetStreamPort}`);
+      }
+
+      try {
+        if (_.isFunction(socket?.setNoDelay)) {
+          socket.setNoDelay(true);
+        }
+
+        return {
+          device,
+          tunnel: {
+            Address: tunnel.Address,
+            RsdPort: tunnel.RsdPort,
+          },
+          packetStreamPort,
+          success: true,
+          socket,
+        };
+      } catch (err) {
+        log.warn(`Could not add device to info server: ${err}`);
+
+        return {
+          device,
+          tunnel: {
+            Address: tunnel.Address,
+            RsdPort: tunnel.RsdPort,
+          },
+          packetStreamPort,
+          success: true,
+          socket,
+        };
+      }
+    } catch (error) {
+      const errorMessage = `Failed to create tunnel for device ${udid}: ${error}`;
+      throw new Error(`❌ ${errorMessage}`);
+    }
+  }
+}
+
+/**
+ */
+async function main() {
+  // Create an instance of TunnelCreator
+  const tunnelCreator = new TunnelCreator();
+  tunnelCreator.setupCleanupHandlers();
+
+  const args = process.argv.slice(2);
+
+  // Extract UDID from command line arguments
+  // If the first argument is provided and is not a flag, use it as the UDID
+  let specificUdid = args.includes('--udid') || args.includes('-u');
+
+  // Handle packet stream base port
+  let packetStreamBasePortArg = args.find((arg) => arg.startsWith('--packet-stream-base-port='));
+  if (packetStreamBasePortArg) {
```

Comment:
I observe some unnecessary duplication here. Can we extract the code for args parsing?